cardano-node/src/Cardano/CLI/Run.hs

```diff
@@ -137,78 +157,139 @@ data ClientCommand
       formatter = F.sformat Crypto.hashHexF . Genesis.unGenesisHash . snd
 
   liftIO . putTextLn $ formatter eGen
 
-runCommand co (PrintSigningKeyAddress netMagic skF) = do
+runCommand co (PrintSigningKeyAddress netMagic skF) _ = do
   sK <- readSigningKey co skF
   let sKeyAddress = prettyAddress . Common.makeVerKeyAddress netMagic $ Crypto.toVerification sK
   liftIO $ putTextLn sKeyAddress
 
-runCommand co (Keygen (NewSigningKeyFile skF) passReq) = do
+runCommand co (Keygen (NewSigningKeyFile skF) passReq) _ = do
   pPhrase <- liftIO $ getPassphrase ("Enter password to encrypt '" <> skF <> "': ") passReq
   sK <- liftIO $ keygen pPhrase
   serDk <- liftIO $ coSerialiseDelegateKey co sK
   liftIO $ ensureNewFileLBS skF serDk
 
-runCommand co (ToVerification skFp (NewVerificationKeyFile vkFp)) = do
+runCommand co (ToVerification skFp (NewVerificationKeyFile vkFp)) _ = do
   sk <- readSigningKey co skFp
   let vKey = Builder.toLazyText . Crypto.formatFullVerificationKey $ Crypto.toVerification sk
   liftIO $ ensureNewFile TL.writeFile vkFp vKey
 
-runCommand co (IssueDelegationCertificate magic epoch issuerSK delegateVK cert) = do
+runCommand co (IssueDelegationCertificate magic epoch issuerSK delegateVK cert) _ = do
   vk <- readVerificationKey delegateVK
   sk <- readSigningKey co issuerSK
   let byGenDelCert :: Delegation.Certificate
       byGenDelCert = issueByronGenesisDelegation magic epoch sk vk
   sCert <- liftIO $ coSerialiseDelegationCert co byGenDelCert
   liftIO $ ensureNewFileLBS (nFp cert) sCert
 
-runCommand _ (CheckDelegation magic cert issuerVF delegateVF) = do
+runCommand _ (CheckDelegation magic cert issuerVF delegateVF) _ = do
   issuerVK <- readVerificationKey issuerVF
   delegateVK <- readVerificationKey delegateVF
   liftIO $ checkByronGenesisDelegation cert magic issuerVK delegateVK
 
-runCommand co (SubmitTx topology fp common) = do
+runCommand co (SubmitTx topology fp common) _ = do
   cc <- liftIO $ mkConfiguration mainnetConfiguration common
   tx <- liftIO $ readByronTx fp
   liftIO $ nodeSubmitTx co topology cc tx
 
-runCommand co (SpendGenesisUTxO (NewTxFile ctTx) ctKey genRichAddr outs common) = do
+runCommand co (SpendGenesisUTxO (NewTxFile ctTx) ctKey genRichAddr outs common) _ = do
   cc <- liftIO $ mkConfiguration mainnetConfiguration common
   sk <- readSigningKey co ctKey
   tx <- liftIO $ issueGenesisUTxOExpenditure co genRichAddr outs cc sk
   liftIO . ensureNewFileLBS ctTx $ serialise tx
 
-runCommand co (SpendUTxO (NewTxFile ctTx) ctKey ins outs common) = do
+runCommand co (SpendUTxO (NewTxFile ctTx) ctKey ins outs common) _ = do
   cc <- liftIO $ mkConfiguration mainnetConfiguration common
   sk <- readSigningKey co ctKey
   gTx <- liftIO $ issueUTxOExpenditure co ins outs cc sk
   liftIO . ensureNewFileLBS ctTx $ serialise gTx
 
+runCommand co
+           (GenerateTxs topology numOfTxs numOfOutsPerTx feePerTx tps sigKeysFiles common)
+           loggingLayer = do
```

Comment:
This should be a short case branch that calls out to the main function.

Reply:
Sorry, didn't understand it...

Reply:
Because we don't want functions that are too long.

`runCommand` went through refactoring to avoid precisely that.

Reply:
Done. Now I just pass `loggingLayer` in my function, all tracers will be created on my side.